### PR TITLE
format nested if

### PR DESCRIPTION
**Pull request by YannanGao-gs**
#### What does this PR do / why is it needed ?

nested if
improve https://github.com/finos/legend-engine/pull/2129

#### Which issue(s) this PR fixes:

Fixes #

### CODE REVIEW

File: legend-engine-core/legend-engine-core-language-pure/legend-engine-language-pure-grammar-api/src/test/java/org/finos/legend/engine/language/pure/grammar/api/test/TestJsonToGrammarApi.java [hunk 1]
```diff
@@ -288,7 +288,8 @@ public void testUnionModelMapping()
                 "  meta::pure::mapping::modelToModel::test::shared::dest::Person[p1]: Pure\n" +
                 "  {\n" +
                 "    ~src meta::pure::mapping::modelToModel::test::shared::src::_Person\n" +
-                "    ~filter $src.fullName->startsWith('Johny')\n" +
+                "    ~filter \n" +
+                "       $src.fullName->startsWith('Johny')\n" +
```
Comment (aziemchawdhary-gs) [inline legend-engine-core/legend-engine-core-language-pure/legend-engine-language-pure-grammar-api/src/test/java/org/finos/legend/engine/language/pure/grammar/api/test/TestJsonToGrammarApi.java:292]: Why has the formatting of filter changed here and below? Isn't this change formatting if statements? I don't see an if statement involved here

File: legend-engine-core/legend-engine-core-language-pure/legend-engine-language-pure-grammar/src/main/java/org/finos/legend/engine/language/pure/grammar/to/DEPRECATED_PureGrammarComposerCore.java [hunk 2]
```diff
@@ -494,22 +516,33 @@ public String visit(PureInstanceClassMapping pureInstanceClassMapping)
         String pureFilter = "";
         if (pureInstanceClassMapping.filter != null)
         {
+            this.currentTabLevel += 1;
             pureInstanceClassMapping.filter.parameters = Collections.emptyList();
-            String filterString = pureInstanceClassMapping.filter.accept(this).replaceFirst("\\|", "");
-            pureFilter = getTabString(2) + "~filter " + filterString + "\n";
+            String filterString = pureInstanceClassMapping.filter.accept(Builder.newInstance().withCurrentTabLevel(this.currentTabLevel + 1).build()).replaceFirst("\\|", "");
+            pureFilter = getTabString(2) + "~filter \n " + getTabString(this.currentTabLevel + 1) + filterString + "\n";
```
Comment (aziemchawdhary-gs) [inline legend-engine-core/legend-engine-core-language-pure/legend-engine-language-pure-grammar/src/main/java/org/finos/legend/engine/language/pure/grammar/to/DEPRECATED_PureGrammarComposerCore.java:522]: Is this intended? Does it make the code clearer to just move the filter down one level and indent?

**Comment by MauricioUyaguari** [comment]
@aziemchawdhary-gs  thanks for the review. this is from https://github.com/finos/legend-engine/pull/2129. thought it was good but had a quick look yesterday and have concerns as well so will look at addressing my concerns and your comments

**Comment by finos-admin** [comment]
This PR is stale because it has been open for 30 days with no activity. Please remove stale label or add any comment to keep this open. Otherwise this will be closed in 5 days.

**Comment by finos-admin** [comment]
This PR was closed because it has been inactive for 35 days. Please re-open if this PR is still relevant.

**Pull request closed without merging**